contracts/factory/src/tests/queries.rs

```diff
@@ -221,4 +222,183 @@ fn test_deploy_multiple_liquidity_pools() {
+        stake_wasm_hash: install_stake_wasm(&env),
+        min_bond: 10i128,
+        max_distributions: 10u32,
+        min_reward: 5i128,
+    };
+
+    let second_token_init_info = TokenInitInfo {
+        token_wasm_hash: install_token_wasm(&env),
+        token_a: token3.clone(),
+        token_b: token4.clone(),
+    };
+    let second_stake_init_info = StakeInitInfo {
+        stake_wasm_hash: install_stake_wasm(&env),
+        min_bond: 5i128,
+        max_distributions: 5u32,
+        min_reward: 2i128,
+    };
+
+    let third_token_init_info = TokenInitInfo {
+        token_wasm_hash: install_token_wasm(&env),
+        token_a: token5.clone(),
+        token_b: token6.clone(),
+    };
+    let third_stake_init_info = StakeInitInfo {
+        stake_wasm_hash: install_stake_wasm(&env),
+        min_bond: 6i128,
+        max_distributions: 6u32,
+        min_reward: 3i128,
+    };
+
+    let lp_wasm_hash = install_lp_contract(&env);
+
+    let first_lp_init_info = LiquidityPoolInitInfo {
+        admin: admin.clone(),
+        fee_recipient: user.clone(),
+        lp_wasm_hash: lp_wasm_hash.clone(),
+        max_allowed_slippage_bps: 5_000,
+        max_allowed_spread_bps: 500,
+        share_token_decimals: 7,
+        swap_fee_bps: 0,
+        token_init_info: first_token_init_info.clone(),
+        stake_init_info: first_stake_init_info,
+    };
+
+    let second_lp_init_info = LiquidityPoolInitInfo {
+        admin: admin.clone(),
+        fee_recipient: user.clone(),
+        lp_wasm_hash: lp_wasm_hash.clone(),
+        max_allowed_slippage_bps: 4_000,
+        max_allowed_spread_bps: 400,
+        share_token_decimals: 6,
+        swap_fee_bps: 0,
+        token_init_info: second_token_init_info,
+        stake_init_info: second_stake_init_info,
+    };
+
+    let third_lp_init_info = LiquidityPoolInitInfo {
+        admin: admin.clone(),
+        fee_recipient: user.clone(),
+        lp_wasm_hash,
+        max_allowed_slippage_bps: 4_000,
+        max_allowed_spread_bps: 400,
+        share_token_decimals: 6,
+        swap_fee_bps: 0,
+        token_init_info: third_token_init_info,
+        stake_init_info: third_stake_init_info,
+    };
+
+    factory.create_liquidity_pool(&first_lp_init_info);
+    factory.create_liquidity_pool(&second_lp_init_info);
+    factory.create_liquidity_pool(&third_lp_init_info);
+
+    let lp_contract_addr = factory.query_pools().get(0).unwrap();
+    let first_result = factory.query_pool_details(&lp_contract_addr);
+    let share_token_addr: Address = env.invoke_contract(
+        &lp_contract_addr,
+        &Symbol::new(&env, "query_share_token_address"),
+        Vec::new(&env),
+    );
+    let first_lp_config: Config = env.invoke_contract(
+        &lp_contract_addr,
+        &Symbol::new(&env, "query_config"),
+        Vec::new(&env),
+    );
+
+    assert_eq!(
+        first_lp_init_info.max_allowed_spread_bps,
+        first_lp_config.max_allowed_spread_bps
+    );
```

Comment:
There's no point in testing those addresses in this particular case. If you keep test cases to essentials then it's easier to read and to not overlook potential errors. 

```suggestion
```